module/plugins/crypter/HoerbuchIn.py

```diff
@@ -35,30 +33,42 @@ def decrypt(self, pyfile):
             html = self.load(pyfile.url)
             soup = BeautifulSoup.BeautifulSoup(html, convertEntities=BeautifulSoup.BeautifulStoneSoup.HTML_ENTITIES)
 
-            abookname = soup.find("a", attrs={'rel': "bookmark"}).text
-            for a in soup.findAll("a", attrs={'href': self.protection}):
-                package = "%s (%s)" % (abookname, a.previousSibling.previousSibling.text[:-1])
-                links = self.decrypt_folder(a['href'])
+            links = []
+            for a in soup.findAll("a", attrs={'href': self.hoster_links}):
+                for decrypted_link in self.decrypt_folder(a.get('href')):
+                    links.append(decrypted_link)
 
-                self.packages.append((package, links, package))
+            self.packages.append((pyfile.name, links, pyfile.name))
         else:
             self.links = self.decrypt_folder(pyfile.url)
 
-
     def decrypt_folder(self, url):
-        m = self.protection.search(url)
+
+        m = self.hoster_links.search(url) or self.protection.search(url)
+
         if m is None:
             self.fail(_("Bad URL"))
         url = m.group(0)
 
-        self.pyfile.url = url
+        if self.hoster_links.match(url):
```

Comment:
This may be done better as I only want the final url from the referral url http://www.hoerbuch.us/wp/goto/Download/34534564